app/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersController.scala

```diff
@@ -110,23 +110,26 @@ class SchemeMemberDetailsAnswersController @Inject()(
 
   def onSubmit(srn: Srn, index: Max300, mode: Mode): Action[AnyContent] =
     identifyAndRequireData(srn).async { implicit request =>
-      lazy val memberDetailsChanged: Boolean = (for {
-        initial <- request.pureUserAnswers
-        initialMemberPayments <- initial.buildMemberDetails(srn, index)
-        currentMemberPayments <- request.userAnswers.buildMemberDetails(srn, index)
-      } yield initialMemberPayments != currentMemberPayments).getOrElse(false)
+      //      val memberDetailsChanged: Boolean = (for {
+      //        initial <- request.pureUserAnswers
+      //        initialMemberPayments <- initial.buildMemberDetails(srn, index)
+      //        currentMemberPayments <- request.userAnswers.buildMemberDetails(srn, index)
+      //      } yield initialMemberPayments != currentMemberPayments).getOrElse(false)
+
+      lazy val memberDetailsChanged =
+        request.userAnswers.changed(_.buildMemberDetails(srn, index))
 
       lazy val justAdded = mode.isNormalMode && request.userAnswers.get(MemberStatus(srn, index)).isEmpty
```

Comment:
Is it maybe that the justAdded flag should be true in NormalMode and CheckMode too?